sdk/cosmos/azure-cosmos-spark_3-0_2-12/src/main/scala/com/azure/cosmos/spark/ChangeFeedInitialOffsetWriter.scala

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.cosmos.spark
+
+import org.apache.commons.io.IOUtils
+import org.apache.spark.sql.SparkSession
+import org.apache.spark.sql.execution.streaming.{HDFSMetadataLog, MetadataVersionUtil}
+
+import java.io.{BufferedWriter, InputStream, InputStreamReader, OutputStream, OutputStreamWriter}
+import java.nio.charset.StandardCharsets
+
+private class ChangeFeedInitialOffsetWriter
+(
+  sparkSession: SparkSession,
+  metadataPath: String
+) extends HDFSMetadataLog[String](sparkSession, metadataPath) {
+
+  val VERSION = 1
+
+  override def serialize(offsetJson: String, out: OutputStream): Unit = {
+    val writer = new BufferedWriter(new OutputStreamWriter(out, StandardCharsets.UTF_8))
+    writer.write(s"v$VERSION\n")
```

Comment:
nit: Would it be worth writing all Json? Not that it would make a functional difference but it might be easier on the parsing maybe?

Reply:
The content shows up in logs etc. - I intentionally kept it at base64 to avoid that customers try to mess around with it. I am not strongly opinionated - obfuscation doesn't really protect from it anyway - but if we change on the format it makes it easier to argue that it is not a breaking change.
For logging/debugging json is helpful thought. By I also created a helper that could be used to log the clear json.